dev/background-api.py

```diff
@@ -0,0 +1,277 @@
+# - build a background model (could be 1D (offset) or 2D (energy, offset) or
+#   2D (X, Y) or 3D (X, Y, energy)) for each off-run bin
+
+# High-level pseudo-code (could actually be code that runs a month from now)
+# for 2D (DETX, DETY):
+# TODO: DETX, DETY not fully implemented in gammapy.data.EventList.
+# TODO: think of other models (1D (offset) or 2D (energy, offset) or 3D
+# (X, Y, energy)).
+
+from gammapy.obs import DataStore
+from gammapy.obs import ObservationTable, RunList
+from gammapy.obs import RunGroups
+from gammapy.background import make_1d_background_model
+
+data_store = DataStore(‘hess’)
+
+# Step 1: select runs to use as off runs & group them (zenith, azimuth
+observation_selection = dict(shape='box', frame='galactic',
+                             lon=(-120, 70), lat=(-5, 5), border=2)
+run_groups = RunGroups(zenith=[0, 20, 40, 60], deselect runs in Galactic plane...)
+# run_groups could be an astropy Table with one row per run
+# and a column RUN_GROUP_ID = 1, 2, 3, ...
+
+# TODO: what about exclusion regions???
+-> gammapy.datasets.load_catalog_tevcat() and cut out circle with radius 0.3 deg.
+-> correct livetime accordingly
+
+parameters = dict(smooth=0.1 deg, …, cutout_agn=True)
+bg_model_set = make_bg_model_set(run_groups, parameters)
+# start of implementation of the make_bg_model_set function
+from group_id in run_groups.ids:
+    event_list_filenames = run_groups.get_event_list_filename(group_id)
+    # offset_bg_hist is an object of class gammapy.data.CountsOffsetHistogram
+counts_xy_hist = histogram_xy(run_lists)
+livetime_xy_hist =
+
+bg_model_set.write(folder_name)
+
+
+# Use case: simulate toy bg
+# - given a certain observation time and observation condition (i.e.
+#   zenith/azimuth angle)
+# - simulate bg events according to a simple analytical model
+# - save the events into an event list (optional: save also binned data)
+# this function can be used for testing different bg modeling methods:
+# does the modeled bg match the analytical model used for the simulations?
+# reference: https://github.com/gammapy/gammapy/issues/1
+
+# High-level pseudo-code:
+
+from gammapy.data import EventList
+from astropy.units import Quantity
+from astropy.coordinates import Angle
+from scipy.integrate import quad
+from root_numpy import random_sample
+from ROOT import TF1
+
+alt = Angle(90 - 30, "deg")
+az = Angle(200, "deg")
+time = Quantity(20, "h")
+
+##bg_event_list = simulate_toy_bg_one_bin(alt, az, time)
+
+# bg model:
+# - spatial coordinates: Gaussian w.r.t. center of FoV
+#   TODO: shouldn't this depend on alt, az?
+#   desirable: use radial aceptance curves, depending on alt, az.
+# - energy: bg spectrum multiplied by the effective area
+# for the bg spectrum one can use the Hegra spectrum:
+#  power-law with index 2.7
+#  ref: http://journals.aps.org/prd/pdf/10.1103/PhysRevD.59.092003
+# alternative: use BESS spectrum for low-E cut-off:
+# http://iopscience.iop.org/0004-637X/545/2/1135
```

Comment:
Note that for cosmic ray spectra we already have the [cosmic_ray_flux](https://gammapy.readthedocs.org/en/latest/api/gammapy.spectrum.cosmic_ray_flux.html) function.

But simulating background as (background spectrum) x (effective area) is not a good approximation, because there's the energy-dependent gamma-hadron separation efficiency missing.
So I'd say, either come up with some toy analytical background model or use the CTA background rate FITS files to simulate background.

Also, below already giving a prototype implementation (calling `quad` or `transform_polar_to_cartesian` or `random_sample`). Instead I'd prefer this API discussion to be higher level, e.g. in this case it could be:
```python
telarray = TelArray('cta-prod2')
observation_table = ... # ObservationTable with some distribution of zenith angles
for observation in observation_table:
    event_list = telarray.simulate_background(observation)
    event_list.write_to_folder('sim_data')
```
I haven't thought about this much, this is not my actual API proposal, it's just to illustrate to you the level at which I think we should discuss here. Having such small code snippets for important use cases will help us figure out which classes and functions we need. Implementation and API details can be listed at the end of this document, or fleshed out in separate issues / pull requests over the coming months.